static/app/views/explore/hooks/useGetSavedQueries.tsx

```diff
@@ -9,23 +9,25 @@ export type SavedQuery = {
   dateUpdated: string;
   end: string;
   environment: string[];
-  fields: string[];
   id: number;
   interval: string;
   lastVisited: string;
-  mode: string;
   name: string;
-  orderby: string;
   projects: number[];
-  query: string;
+  query: Array<{
```

Comment:
If we could define this as a non-empty array, would that get around the filtering you need to do considering your data model now enforces a query in the array? We could introduce a generic type to use

e.g. found this: https://dev.to/martinpersson/type-safety-with-non-empty-arrays-in-typescript-k78#:~:text=TypeScript%20gives%20us%20the%20ability,Non%2Dempty%20array%22%20type.&text=Here%20we%20create%20a%20generic,it%20could%20also%20include%20more.

Reply:
Thanks, that helps!